File: Leetcode/2180 Count Integers With Even Digit Sum.py
class Solution:
    def countEven(self, num: int) -> int:
        return sum(map(lambda _: 1, filter(is_digit_sum_even, range(2, num + 1))))


def digit_sum(n):
    return sum(map(int, str(n)))


def is_digit_sum_even(n):
    return digit_sum(n) % 2 == 0

File: Leetcode/2180 Count Integers With Even Digit Sum.rs
impl Solution {
    pub fn count_even(num: i32) -> i32 {
        (2..=num)
            .filter(is_digit_sum_even)
            .count() as i32
    }
}

fn is_digit_sum_even(n: &i32) -> bool {
    let digit_sum: i32 = 
        n.to_string()
         .chars()
         .filter_map(|c| c.to_digit(10))
         .map(|d| d as i32)
         .sum();
    digit_sum % 2 == 0
}